### PR TITLE
Take CAPM3 Metadata and IPAM feature into use for target cluster

### PR DESCRIPTION
**Pull request by maelk**
Since IPAM and Metadata were implemented in CAPM3, they could be used in metal3-dev-env to set the IP address statically to test the feature.

### CODE REVIEW

File: lib/network.sh [hunk 1]
```diff
@@ -76,7 +77,11 @@ fi
 # Calculate DHCP range
 network_address dhcp_range_start "$PROVISIONING_NETWORK" 10
 network_address dhcp_range_end "$PROVISIONING_NETWORK" 100
+network_address PROVISIONING_POOL_RANGE_START "$PROVISIONING_NETWORK" 100
+network_address PROVISIONING_POOL_RANGE_END "$PROVISIONING_NETWORK" 200
```
Comment (jan-est) [inline lib/network.sh:81]: Are these ranges only for the cases where user does not define them?

File: vm-setup/roles/v1aX_integration_test/vars/main.yml [hunk 2]
```diff
@@ -17,6 +17,18 @@ KUBERNETES_VERSION: "{{ lookup('env', 'KUBERNETES_VERSION') | default('v1.18.0',
 KUBERNETES_BINARIES_VERSION: "{{ lookup('env', 'KUBERNETES_BINARIES_VERSION') | default(lookup('env', 'KUBERNETES_VERSION') | default('v1.18.0', true), true) }}"
 KUBERNETES_BINARIES_CONFIG_VERSION: "{{ lookup('env', 'KUBERNETES_BINARIES_CONFIG_VERSION') | default('v0.2.7', true) }}"
 
+IP_STACK: "{{ lookup('env', 'IP_STACK') | default('v4', true) }}"
+PROVISIONING_POOL_RANGE_START: "{{ lookup('env', 'PROVISIONING_POOL_RANGE_START') | default('172.22.0.100', true) }}"
+PROVISIONING_POOL_RANGE_END: "{{ lookup('env', 'PROVISIONING_POOL_RANGE_END') | default('172.22.0.200', true) }}"
+PROVISIONING_CIDR: "{{ lookup('env', 'PROVISIONING_CIDR') | default('24', true) }}"
+BAREMETALV4_POOL_RANGE_START: "{{ lookup('env', 'BAREMETALV4_POOL_RANGE_START') | default('192.168.111.100', true) }}"
+BAREMETALV4_POOL_RANGE_END: "{{ lookup('env', 'BAREMETALV4_POOL_RANGE_END') | default('192.168.111.100', true) }}"
```
Comment (furkatgofurov7) [inline vm-setup/roles/v1aX_integration_test/vars/main.yml:25]: should `default` be '192.168.111.200' for `BAREMETALV4_POOL_RANGE_END`?

Comment (jan-est) [inline vm-setup/roles/v1aX_integration_test/vars/main.yml:25]: Should BAREMETALV4_POOL_RANGE_START and BAREMETALV4_POOL_RANGE_END be both defaulted default('192.168.111.100', true)?

Comment (maelk) [reply vm-setup/roles/v1aX_integration_test/vars/main.yml:25]: good catch

File: lib/network.sh (hunk 1 above)
Comment (maelk) [reply lib/network.sh:81]: no, they are always defined

File: vm-setup/roles/v1aX_integration_test/vars/main.yml (hunk 2 above)
Comment (maelk) [reply vm-setup/roles/v1aX_integration_test/vars/main.yml:25]: fixing it

**Comment by maelk** [comment]
The v1a4 tests passed previously and the change made since then is not impacting (defaulting a variable to a different value, since the variable is anyways defined)

File: lib/network.sh [hunk 3]
```diff
@@ -115,7 +110,50 @@ if [[ "${EPHEMERAL_CLUSTER}" == "minikube" ]] && [[ -n "${EXTERNAL_SUBNET_V6}" ]
...
+  export EXTERNAL_SUBNET_V6_PREFIX
+  if [[ -z "${EXTERNAL_SUBNET_V6_HOST}" ]]; then
+    network_address EXTERNAL_SUBNET_V6_HOST "$EXTERNAL_SUBNET_V6" 1
+  fi
+
+  # Calculate DHCP range for baremetal network (20 to 60 is the libvirt dhcp) IPv6
+  network_address BAREMETALV6_POOL_RANGE_START "$EXTERNAL_SUBNET_V6" 100
+  network_address BAREMETALV6_POOL_RANGE_END "$EXTERNAL_SUBNET_V6" 200
+  export BAREMETALV6_POOL_RANGE_START
+  export BAREMETALV6_POOL_RANGE_END
```
Comment (felixest) [inline lib/network.sh:144]: After network_address or prefixlen are called, the variables are already exported. Some exports may be unnecessary here

**Pull request merged**